shell/shared/grain.js

```diff
@@ -273,32 +273,40 @@ Meteor.methods({
       var identity = globalDb.getIdentity(identityId);
       var sharerDisplayName = identity.profile.name;
       var outerResult = {successes: [], failures: []};
-      emailAddresses.forEach(function(emailAddress) {
-        var result = SandstormPermissions.createNewApiToken(
-          globalDb, {identityId: identityId, accountId: accountId}, grainId,
-          "email invitation for " + emailAddress,
-          roleAssignment, {webkey: {forSharing: true}});
-        var url = origin + "/shared/" + result.token;
-        var html = message.html + "<br><br>" +
-            "<a href='" + url + "' style='display:inline-block;text-decoration:none;" +
-            "font-family:sans-serif;width:200px;min-height:30px;line-height:30px;" +
-            "border-radius:4px;text-align:center;background:#428bca;color:white'>" +
-            "Open Shared Grain</a><div style='font-size:8pt;font-style:italic;color:gray'>" +
-            "Note: If you forward this email to other people, they will be able to access " +
-            "the share as well. To prevent this, remove the button before forwarding.</div>";
-        try {
-          SandstormEmail.send({
-            to: emailAddress,
-            from: "Sandstorm server <no-reply@" + HOSTNAME + ">",
-            subject: sharerDisplayName + " has invited you to join a grain: " + title,
-            text: message.text + "\n\nFollow this link to open the shared grain:\n\n" + url +
-              "\n\nNote: If you forward this email to other people, they will be able to access " +
-              "the share as well. To prevent this, remove the link before forwarding.",
-            html: html,
-          });
-          outerResult.successes.push(emailAddress);
-        } catch (e) {
-          outerResult.failures.push({email: emailAddress, error: e.toString()});
+      contacts.forEach(function(contact) {
+        if (contact.isDefault && contact.profile.service === "email") {
```

Comment:
I guess this means we can't yet share-by-identity to an email identity? That seems fine for now, but I think we'll eventually want to make a distinction here.